Gruntfile.js

```diff
@@ -31,11 +31,58 @@ module.exports = function(grunt) {
           urls: ["http://localhost:8512/tests/SpecRunner.html"]
         }
       }
-    }
+    },
+    browserify: {
+      // browserify src/resolver.js -o dist/amd-resolver.js -s amd-resolver --dg false
+      options: {
+        browserifyOptions: {
+          detectGlobals: false,
+          standalone: "amd-resolver"
+        }
+      },
+      debug: {
+        options: {
+          browserifyOptions: {
+            debug: true
+          }
+        },
+        src: "src/resolver.js",
+        dest: "dist/amd-resolver.js"
+      },
+      release: {
+        src: "src/resolver.js",
+        dest: "dist/amd-resolver.js"
+      }
+    },
+    watch: {
+      browserify: {
+        files: "src/**/*.js",
+        tasks: "build-debug"
+      }
+    },
+    jshint: {
+      all: [
+        "Gruntfile.js",
+        "src/**/*.js",
+        "tests/**/*.js"
+      ],
+      options: {
+        reporter: require("jshint-stylish"),
+        jshintrc: true
+      }
+    },
   });
 
   grunt.loadNpmTasks("grunt-mocha");
+  grunt.loadNpmTasks("grunt-browserify");
   grunt.loadNpmTasks("grunt-contrib-connect");
+  grunt.loadNpmTasks("grunt-contrib-watch");
+  grunt.loadNpmTasks("grunt-contrib-jshint");
+
   grunt.registerTask("server", ["connect:keepalive"]);
   grunt.registerTask("test", ["connect:test", "mocha:test"]);
+  grunt.registerTask("lint", ["jshint"]);
+  grunt.registerTask("build-debug", ["jshint", "browserify:debug"]);
+  grunt.registerTask("build-release", ["jshint", "browserify:release"]);
```

Comment:
JSHint is going to run first in these cases to early-fail if there are issues with the code that is being linted.